tests/mappers/test_exon_genomic_coords.py

```diff
@@ -858,309 +1052,274 @@ async def test_transcript_to_genomic(
     tpm3_exon1_exon8,
     ntrk1_exon10_exon17,
 ):
-    """Test that transcript_to_genomic_coordinates works correctly."""
+    """Test that tx_segment_to_genomic works correctly."""
     # TPM3
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
+    resp = await test_egc_mapper.tx_segment_to_genomic(
         exon_start=None, exon_end=8, transcript="NM_152263.3"
     )
-    genomic_data_assertion_checks(resp, tpm3_exon8_g)
+    genomic_tx_seg_service_checks(resp, tpm3_exon8_g)
 
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
+    resp = await test_egc_mapper.tx_segment_to_genomic(
         exon_start=1, exon_end=None, transcript="NM_152263.3"
     )
-    genomic_data_assertion_checks(resp, tpm3_exon1_g)
-
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
-        exon_start=None, exon_end=8, transcript="NM_152263.3       "
-    )
-    genomic_data_assertion_checks(resp, tpm3_exon8_g)
+    genomic_tx_seg_service_checks(resp, tpm3_exon1_g)
 
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
+    resp = await test_egc_mapper.tx_segment_to_genomic(
         exon_start=None, exon_end=8, gene="TPM3", transcript="NM_152263.3"
     )
-    genomic_data_assertion_checks(resp, tpm3_exon8_g)
+    genomic_tx_seg_service_checks(resp, tpm3_exon8_g)
 
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
-        exon_start=None, exon_end=8, gene=" TPM3 ", transcript=" NM_152263.3 "
-    )
-    genomic_data_assertion_checks(resp, tpm3_exon8_g)
-
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
+    resp = await test_egc_mapper.tx_segment_to_genomic(
         exon_start=None, exon_end=8, gene="tpm3", transcript="NM_152263.3"
     )
-    genomic_data_assertion_checks(resp, tpm3_exon8_g)
-
-    expected = copy.deepcopy(tpm3_exon1_exon8)
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
-        exon_start=1, exon_end=8, exon_end_offset=-5, transcript="NM_152263.3"
-    )
-    expected.exon_end = 8
-    expected.exon_end_offset = -5
-    expected.end = 154170405
-    genomic_data_assertion_checks(resp, expected)
+    genomic_tx_seg_service_checks(resp, tpm3_exon8_g)
 
-    resp = await test_egc_mapper.transcript_to_genomic_coordinates(
+    expected = tpm3_exon1_exon8.model_copy(deep=True)
+    resp = await test_egc_mapper.tx_segment_to_genomic(
         exon_start=1, exon_end=8, exon_end_offset=5, transcript="NM_152263.3"
     )
-    expected.exon_end_offset = 5
-    expected.end = 154170395
-    genomic_data_assertion_checks(resp, expected)
+    expected.seg_end.offset = 5
+    expected.seg_end.genomic_location.start = 154170474
```

Comment:
i think this would be `154170404`since we would have to compare to the `alt_start_i` value of `154170399`